File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-2.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        if len(s) == 0: return 0
        numCharMap = {}
        left: int = 0
        numCharMap[s[left]] = 1
        replaceTime: int = 1
        res: int = 0
        for right in range(1, len(s)):
            maxSame = 0
            charNow = s[right]
            if charNow in numCharMap.keys():
                numCharMap[charNow] += 1
            else:
                numCharMap[charNow] = 1
            
            for v in numCharMap.values():
                if v > maxSame:
                    maxSame = v
            
            windowsLength = right - left + 1
            if windowsLength - maxSame > k:
                charLeft = s[left]
                numCharMap[charLeft] -= 1
                left += 1
            if right - left + 1> res:
                res = right - left + 1
        return res

            

                

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.java
class Solution {
    public int findMin(int[] nums) {
        int left = 0;
        int right = nums.length - 1;
        int middle;
        int min = 1001;

        while(left <= right){
            middle = (left + right) / 2;
            if(nums[left] <= nums[middle]){
                if(min > nums[left]) min = nums[left];
                left = middle + 1;
            } else {
                if(min > nums[middle]) min = nums[middle];
                right = middle - 1;
            }
        }
        return min;
    }
}
